dhis-2/dhis-web-api/src/main/java/org/hisp/dhis/webapi/security/config/DhisWebApiWebSecurityConfig.java

```diff
@@ -378,18 +382,37 @@ private void configureAccessRestrictions(
                 .anyRequest().authenticated();
         }
 
+        /**
+         * This method configures almost everything security related to /api
+         * endpoints
+         */
         protected void configure( HttpSecurity http )
             throws Exception
         {
             String[] activeProfiles = getApplicationContext().getEnvironment().getActiveProfiles();
 
+            // Special handling if we are running in embedded Jetty mode
             if ( Arrays.asList( activeProfiles ).contains( "embeddedJetty" ) )
             {
                 // This config will redirect unauthorized requests to standard
                 // http basic (pop-up login form)
                 http.antMatcher( "/**" )
                     .authorizeRequests( this::configureAccessRestrictions )
                     .httpBasic();
+
+                /**
+                 * Setup session handling, this is handled in @see
+                 * DhisWebCommonsWebSecurityConfig when running standard mode
+                 */
+                http
+                    .sessionManagement()
+                    .sessionFixation().migrateSession()
+                    .sessionCreationPolicy( SessionCreationPolicy.ALWAYS )
+                    .enableSessionUrlRewriting( false )
+                    .maximumSessions( 10 )
```

Comment:
Hm...

Reply:
Yes, your suspicion is right. This is not strictly related to this PR. This is an embedded Jetty thing, and might need its separate PR.

Reply:
Mostly I wonder about the hard coded `10` here - only applies for local jetty?

Reply:
Ah, good catch. This is not usually defined, so you are right we should probably not change it. I was just thinking that no normal user should be allowed to have more concurrent sessions. The default in Spring is to allow any number. I think in general you would not want that for normal web users, not sure why this is default. Anyway no way to change it only one place now, so I will change it do the current default behviour.